src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -2611,6 +2611,103 @@ Task Synchronization API can be used for multiple dynamic synchronization patter
+    the operation ready for signal automatically removes it, thus the job is ready again to receive and handle the same signal.
+
+  - Send signal(s): adds one or more signals given as input to the current job signals.
+   This operation removes the corresponding ready signals if they already exist. For instance, sending `soft_stop` removes the signal
+   `ready_soft_stop` if it already exists among the job signals.
+
+  - Check for signal(s) reception: verifies whether one or more signals exist among the set of job signals.
+
+  - Wait for signal(s): blocking instruction that waits until one signal (among those given as input) is added to the set of job signals.
+
+  - Remove signal(s): removes one or more signals from the set of job signals.
+
+In addition to the above operations available within ProActive Tasks, users and third-party systems can send signals
+to a specific job via link:/rest/#1218178377[ProActive Signals REST endpoint].
+
+==== Examples
+Two buckets in ProActive *Catalog* contain workflow examples and templates, which illustrate the usage of the Signal API, namely:
+
+* The *Signal* bucket contains many workflow examples that use all signal operations.
+These workflows use a set of predefined groovy scripts stored in the same bucket. Additionally, below we give an example of a groovy code that uses some Signal API operations.
+
+[source, groovy]
+----
+//Declare that the current job is ready to receive and handle the 'soft_stop' signal
+signalapi.readyForSignal("soft_stop")
+
+//Display current job signals
+println("Job signals "+ signalapi.getJobSignals())
+
+//Wait util the 'soft_stop' signal is sent (via the Workflow Execution portal or REST API)
+signalapi.waitFor("soft_stop")
+
+//Remove the 'soft_stop' signal from the set of job signals
+signalapi.removeSignal("soft_stop")
+----
+
+* The *Controls* bucket contains two workflow templates, viz.,:
+
+  - _Check_For_Signals_: a control pattern that executes a loop until one ore more signals given as input are received.
+
+  - _Wait_For_Signals_: a blocking control pattern that blocks the task execution waiting for the reception of one or more signals given as input.
+
+Concerning the usage of the link:/rest/#1218178377[Signals REST endpoint], below we provide an example using curl that shows how to send a signal to a specific job:
+[source,shell,subs="verbatim,quotes"]
+----
+curl -X POST -H "sessionid:$sessionid"
+http://try.activeeon.com:8080/rest/scheduler/job/`job_id`/signals?signal="*soft_stop*"
+----
+==== Conclusion
+The Signal API allows users to interact with jobs during their execution, thus enabling to impact and change jobs' behaviors at runtime.
+Many use cases can be fulfilled using this API. For instance:
+
+  - _Softly stopping a job_: when the stop signal is received, the job executes some code that properly finish this job.
+
+  - _Scaling up and down resources on the user demand_: a job that deploys resources (e.g., compute resources, workers/agents, etc)
```

Comment:
I don't understand this example. A job that deploys resources, what is that ? I don't see it as a "typical usage of the API".

Why not talking rather about a loop with continue or stop features ?

Reply:
I added the use case you mentioned, but also kept the mine :)